app/src/main/java/com/pileproject/drive/execution/RxObservableProgram.java

```diff
@@ -69,7 +69,7 @@
     public static final int MESSAGE_RESTARTED = 4;
     public static final int MESSAGE_BLOCK_EXECUTED = 5;
 
-    private final MachineController mMachineController;
+    private final CarControllerBase mMachineController;
```

Comment:
Here, this forces us to use `CarControllerBase`. It may limit the extensibility.